plugin/stripcomments/stripcomments.go

```diff
@@ -0,0 +1,105 @@
+package stripcomments
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"slices"
+
+	"github.com/safedep/code/core"
+	"github.com/safedep/dry/log"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type StripCommentsCallback func(f core.File, r io.Reader) error
+
+type stripCommentsPlugin struct {
+	stripCommentsCallback StripCommentsCallback
+}
+
+// Verify contract
+var _ core.TreePlugin = (*stripCommentsPlugin)(nil)
+
+func NewStripCommentsPlugin(stripCommentsCallback StripCommentsCallback) *stripCommentsPlugin {
+	return &stripCommentsPlugin{
+		stripCommentsCallback: stripCommentsCallback,
+	}
+}
+
+func (p *stripCommentsPlugin) Name() string {
+	return "StripCommentsPlugin"
+}
+
+var supportedLanguages = []core.LanguageCode{core.LanguageCodePython, core.LanguageCodeJavascript}
+
+func (p *stripCommentsPlugin) SupportedLanguages() []core.LanguageCode {
+	return supportedLanguages
+}
+
+func (p *stripCommentsPlugin) AnalyzeTree(ctx context.Context, tree core.ParseTree) error {
+	lang, err := tree.Language()
+	if err != nil {
+		return fmt.Errorf("failed to get language: %w", err)
+	}
+
+	file, err := tree.File()
+	if err != nil {
+		return fmt.Errorf("failed to get file: %w", err)
+	}
+
+	treeData, err := tree.Data()
+	if err != nil {
+		return fmt.Errorf("failed to get tree data: %w", err)
+	}
+
+	log.Debugf("stripcomments - Analyzing tree for language: %s, file: %s\n",
+		lang.Meta().Code, file.Name())
+
+	// @TODO - If performance issues arise due to copying the entire file contents into memory,
+	// consider storing in a temporary file and returning a reader to that file
+	var output bytes.Buffer
+	stripComments(tree.Tree().RootNode(), *treeData, &output)
+
+	return p.stripCommentsCallback(file, &output)
+}
+
+func stripComments(node *sitter.Node, source []byte, output *bytes.Buffer) {
+	if node == nil {
+		return
+	}
+
+	// Skip comments and standalone doc strings
+	if node.Type() == "comment" || (node.Type() == "string" && isStandaloneDocstring(node)) {
```

Comment:
Should we have checks like `isJavascriptCommentNode()`, `isPythonCommentNode()` and encapsulate language specific evaluations within such functions. It will improve readability and also allow us to easily add support for more languages.

Reply:
yep, this is better,
we can just have a `isCommentNode(node, lang)` and then it'll take care of it internally
we can have a map internally with checks for every language specifically